src/lib/cache/gdfs_cache.hpp

```diff
@@ -1,19 +1,23 @@
 #pragma once
 
 #include <list>
+#include <shared_mutex>
 #include <unordered_map>
 #include <utility>
 
-#include "abstract_cache_impl.hpp"
+#include "abstract_cache.hpp"
 #include "boost/heap/fibonacci_heap.hpp"
 #include "utils/assert.hpp"
 
 namespace opossum {
 
-// Generic cache implementation using the GDFS policy.
-// Note: This implementation is not thread-safe.
+/*
+ * Generic cache implementation using the GDFS policy.
+ * Note: This implementation is only thread-safe for basic cache operations.
```

Comment:
Not sure if this comments helps a lot. What is a "basic" cache operation?

Reply:
I kind of answered this below, this means (try-)get, set, has, and snapshot. If I add locks to the other, this will be obsolete.